fix: Migration script should also work for repos configured as "forks"

pwnage101 opened the pull request:
Previously, I had assumed all repos had an "origin" remote. This is a fatal assumption because several enterprise repos had been included in FORKED_REPOS on newer devstacks so they were configured without an "origin" remote. This fixes the migration script to handle the fork repo case---instead of re-writing the "origin" remote, re-point the main branch to track the "edx" remote instead of the "openedx" remote.

ENT-11240 (epic: ENT-11239)

Testing
---

From devstack:
```
$ make migrate-repo-git-to-edx
./migrate-repo-git-to-edx.sh
Updating enterprise-access ...
Old tracking remote: openedx
New tracking remote: edx

Updating enterprise-subsidy ...
Old tracking remote: openedx
New tracking remote: edx

Updating enterprise-catalog ...
Old tracking remote: openedx
New tracking remote: edx

Updating license-manager ...
Old tracking remote: openedx
New tracking remote: edx

Migration complete.
```

That resulted in all 4 enterprise IDA checkouts to have their main/master branches re-target the "edx" remote. Re-running the command is idempotent:

```
$ make migrate-repo-git-to-edx
./migrate-repo-git-to-edx.sh
Updating enterprise-access ...
Skipping enterprise-access: main branch was already configured to track edx remote.
Updating enterprise-subsidy ...
Skipping enterprise-subsidy: main branch was already configured to track edx remote.
Updating enterprise-catalog ...
Skipping enterprise-catalog: master branch was already configured to track edx remote.
Updating license-manager ...
Skipping license-manager: master branch was already configured to track edx remote.
Migration complete.
```

mikedikan commented:
FYI @edx/docker-devstack-working-group

robrap commented on README.rst:
```diff
@@ -543,6 +543,20 @@ up the provisioning process on Mac), so you can try the following:
+   ...
+   Build failed running pavelib.assets.update_assets: Subprocess return code: 137
+
+This is an indication that your docker process ran out of memory during execution.
```
I'd note that return code 137 basically means some went majorly wrong and died.  We have found that this often can be fixed by adjusting memory, but if that doesn't work, more sleuthing will be required.

mikedikan replied:
comment (I believe) addressed - please take another look :)

robrap commented on README.rst:
```diff
@@ -543,6 +543,21 @@ up the provisioning process on Mac), so you can try the following:
+   ...
+   Build failed running pavelib.assets.update_assets: Subprocess return code: 137
+
+This is an indication that your docker process died during execution.  Most likely,
```
Nit: What does it actually kill?  I'm unclear if it is the original docker process that dies, or just the additional command you ran, etc.?

mikedikan replied:
The entire process (make provision) dies, resulting in a exit in the middle of provisioning (very annoying, as we need to restart provisioning)  Should I try to make that more clear?  Suggested language would be helpful if so.

robrap replied:
Just wondering.  I don't think it needs any changes.  Someone will know what it means when they see the error. :)

staubina replied:
@robrap You can get this error in any container as a note.  I had this error in my MySql container when I ran into this issue.

robrap commented on README.rst:
```diff
@@ -543,6 +543,21 @@ up the provisioning process on Mac), so you can try the following:
+
+This is an indication that your docker process died during execution.  Most likely,
+this is due to running out of memory (this has historically been the case at edX).
+If you are set to 2GB (docker for mac default) up it to 4GB (the edX recommendation).
```
Some minor changes I would make are detailed here:
```
... Most likely, this is due to running out of memory. If you are set to 2GB (docker for mac default), increase it to 4GB (the current recommendation). If you are set to 4GB, try 6GB.
```

mikedikan replied:
Sounds like you want to strip out the personality here 😝

robrap replied:
I like the personality, but am concerned for people where english might be their second language.  I also removed references to "edX" that weren't needed, or could have been "Open edX", yadda yadda yadda.

pwnage101 committed:
fix: Migration script should also work for repos configured as "forks"

Previously, I had assumed all repos had an "origin" remote. This is a
fatal assumption because several enterprise repos had been included in
FORKED_REPOS on newer devstacks so they were configured without an
"origin" remote. This fixes the migration script to handle the fork repo
case---instead of re-writing the "origin" remote, re-point the main
branch to track the "edx" remote instead of the "openedx" remote.

ENT-11240 (epic: ENT-11239)
